### PR TITLE
[FRR]: Update frr to frr-7.0.1

### PR DESCRIPTION
**Pull request by pavel-shirshov**
**PLEASE DON"T MERGE YET!**

**- What I did**
1. Updated FRR to version FRR 7.0.1.
2. Enabled staticd, which controls static routes since FRR 6.x
3. Disabled installation of FRR-pythontools, which contains only one python tool, which makes semantic diffs of quagga configs, but requires python3 to run
4. Fixed some issues in Makefiles

**- How I did it**
Added necessary dependency + updated FRR repository

**- How to verify it**
Build an image and run it.

**- Description for the changelog**

Update FRR to FRR 7.0.1. staticd was enabled.

** - What must be completed until the PR is ready for merging: **
1. Convert all Debian related downloads to azure cloud related downloads [DONE]
2. Decide what to do with frr.conf vs bgpd.conf,zebra.conf,staticd.conf [DONE]
3. Update permissions for /etc/frr. Currently it's impossible for frr create a new file [DONE]
```
# wr mem
Note: this version of vtysh never writes vtysh.conf

Warning: attempting direct configuration write without watchfrr.
File permissions and ownership may be incorrect, or write may fail.

Building Configuration...
Warning: /etc/frr/frr.conf.sav unlink failed
Integrated configuration saved to /etc/frr/frr.conf
[OK]
```

### CODE REVIEW

**Comment by lguohan** [comment]
you need to modify the start.sh to start staticd

**Comment by lguohan** [comment]
both model are supported, if it is unified configuration model, then use frr.conf. if it is separated configuration model, then use zebra.conf, bgpd.conf, and staticd.conf.

**Comment by lguohan** [comment]
about FRR-pythontools, don't we have python3 installed in the frr docker?

File: src/python3/Makefile [hunk 1]
```diff
@@ -15,6 +15,7 @@ DERIVED_TARGETS = lib$(PYTHON_PNAME)-stdlib_$(PYTHON_VER)-$(PYTHON_DEB_VER)_amd6
 				  #$(PYTHON_PNAME)-dev_$(PYTHON_VER)-$(PYTHON_DEB_VER)_amd64.deb
 
 $(addprefix $(DEST)/, $(MAIN_TARGET)): $(DEST)/% :
+	rm -fr $(PYTHON_PNAME)-$(PYTHON_VER)
```
Comment (qiluo-msft) [inline src/python3/Makefile]: >rm -fr $(PYTHON_PNAME)-$(PYTHON_VER) [](start = 1, length = 36)

This fix could be separated into standalone PR and complete quickly.

**Comment by qiluo-msft** [comment]
After this PR [https://github.com/Azure/sonic-buildimage/pull/2807](https://github.com/Azure/sonic-buildimage/pull/2807), python3 is only available in snmp docker image (3.6). What is the motivate of using FRR-pythontools? If really needed, I propose 2 options:
1. install python3.6 into docker-base image
2. install Debian vanilla python3.5.3-1 in all Stretch docker-base.

---
In reply to: [492426979](https://github.com/Azure/sonic-buildimage/pull/2899#issuecomment-492426979) [](ancestors = 492426979)

**Comment by paulnice** [comment]
> you need to modify the start.sh to start staticd

I think I've it here: https://github.com/Azure/sonic-buildimage/pull/2899/files#diff-2faaa492fd949e4b9f384eb592a81bcbR35

**Comment by pavel-shirshov** [comment]
> you need to modify the start.sh to start staticd

I think I've it here: https://github.com/Azure/sonic-buildimage/pull/2899/files#diff-2faaa492fd949e4b9f384eb592a81bcbR35

**Comment by pavel-shirshov** [comment]
> After this PR #2807, python3 is only available in snmp docker image (3.6). What is the motivate of using FRR-pythontools? If really needed, I propose 2 options:
> 
> install python3.6 into docker-base image
> install Debian vanilla python3.5.3-1 in Stretch docker-base.
> 
> 
> In reply to: 492426979 [](ancestors = 492426979)

I don't think we need frr-pythontools.
Currently it's contains only https://github.com/Azure/sonic-frr/blob/88351c8f6df5450e9098f773813738f62abb2f5e/tools/frr-reload.py
I think we can run in on python2.7 and we can install it inside of dockerfile if we need it.

**Comment by pavel-shirshov** [comment]
We need more free space for our Jenkins

**Comment by pavel-shirshov** [comment]
> both model are supported, if it is unified configuration model, then use frr.conf. if it is separated configuration model, then use zebra.conf, bgpd.conf, and staticd.conf.

Currently frr.conf has preference over bgpd.conf/zebra.com/staticd.conf

File: src/python3/Makefile (hunk 1 above)
Comment (pavel-shirshov) [reply src/python3/Makefile]: ok

File: src/libyang/Makefile [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+SHELL = /bin/bash
+.SHELLFLAGS += -e
+
+MAIN_TARGET = $(LIBYANG)
+DERIVED_TARGETS = $(LIBYANG_DEV)
+DEPS = $(DEST)/$(SWIG_BASE) $(DEST)/$(SWIG)
+
+$(addprefix $(DEST)/, $(MAIN_TARGET)): $(DEST)/% :
+	rm -fr ./libyang-$(LIBYANG_VERSION)
+	sudo dpkg -i $(DEPS)
```
Comment (pavel-shirshov) [inline src/libyang/Makefile]: I need an advice here. I don't know why, but I need to use sudo dpkg -i here, just dependency was not enough.

File: rules/libyang.mk [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+LIBYANG_VERSION = $(LIBYANG_VERSION_BASE).105
+LIBYANG_SUBVERSION = 1
+
+export LIBYANG_VERSION_BASE
+export LIBYANG_VERSION
+export LIBYANG_SUBVERSION
+
+LIBYANG = libyang$(LIBYANG_VERSION_BASE)_$(LIBYANG_VERSION)-$(LIBYANG_SUBVERSION)_amd64.deb
+$(LIBYANG)_SRC_PATH = $(SRC_PATH)/libyang
+$(LIBYAND)_DEPENDS += $(SWIG_BASE) $(SWIG)
```
Comment (lguohan) [inline rules/libyang.mk]: LIBYANG

File: src/libyang/Makefile (hunk 2 above)
Comment (lguohan) [reply src/libyang/Makefile]: you have a typo in your libyang rule file.

File: sonic-slave-stretch/Dockerfile [hunk 4]
```diff
@@ -109,7 +114,6 @@ RUN apt-get update && apt-get install -y \
         dpatch \
         libdb-dev \
         iptables-dev \
-        swig \
```
Comment (lguohan) [inline sonic-slave-stretch/Dockerfile]: this will break the swss common build, can you add swig into the swss_common build depends.

File: dockers/docker-fpm-frr/supervisord.conf [hunk 5]
```diff
@@ -39,6 +39,15 @@ startsecs=0
...
 stderr_logfile=syslog
 
+[program:staticd]
+command=/usr/lib/frr/staticd -A 127.0.0.1
+priority=4
+autostart=false
+autorestart=false
+startsecs=0
+stdout_logfile=syslog
+stderr_logfile=syslog
```
Comment (nazariig) [inline dockers/docker-fpm-frr/supervisord.conf]: @lguohan @pavel-shirshov can we force this fix with a separate PR?

Comment (pavel-shirshov) [reply dockers/docker-fpm-frr/supervisord.conf]: Why?

**Comment by lguohan** [comment]
retest this please

File: dockers/docker-fpm-frr/supervisord.conf (hunk 5 above)
Comment (nazariig) [reply dockers/docker-fpm-frr/supervisord.conf]: @pavel-shirshov our regression fails because frr is missing staticd which is essential part of this suite

Comment (pavel-shirshov) [reply dockers/docker-fpm-frr/supervisord.conf]: @nazarii-gnydyn 
Can you please share some output?
On our side I see: 
```
# ps ax
  PID TTY      STAT   TIME COMMAND
    1 pts/0    Ss+    0:01 /usr/bin/python /usr/bin/supervisord
   25 pts/0    Sl     0:00 /usr/sbin/rsyslogd -n
   30 pts/0    S      0:00 python /usr/bin/bgpcfgd
   33 pts/0    Sl     0:08 /usr/lib/frr/zebra -A 127.0.0.1 -s 90000000 -M fpm
   36 pts/0    S      0:00 /usr/lib/frr/staticd -A 127.0.0.1
   39 pts/0    Sl     0:02 /usr/lib/frr/bgpd -A 127.0.0.1
   47 pts/0    Sl     0:27 fpmsyncd
   49 pts/1    Ss     0:00 bash
   65 pts/1    R+     0:00 ps ax
```

Comment (nazariig) [reply dockers/docker-fpm-frr/supervisord.conf]: @pavel-shirshov On our side (without this fix):
root@sonic:/home/admin# docker exec -ti bgp bash
root@sonic:/# ps -aux
USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root         1  0.5  0.2  58600 20444 pts/0    Ss+  13:03   0:00 /usr/bin/python /usr/bin/supervisord
root        29  0.0  0.0 250136  2860 pts/0    Sl   13:03   0:00 /usr/sbin/rsyslogd -n
root        34  0.0  0.1  45556 13788 pts/0    S    13:03   0:00 python /usr/bin/bgpcfgd
frr         37  0.1  0.0 298552  5884 pts/0    Sl   13:03   0:00 /usr/lib/frr/zebra -A 127.0.0.1 -s 90000000 -M fpm
frr         39  0.1  0.1 299900  8964 pts/0    Sl   13:03   0:00 /usr/lib/frr/bgpd -A 127.0.0.1
root        45  0.5  0.0 101356  4096 pts/0    Sl   13:03   0:00 fpmsyncd
root        47  2.5  0.0  18184  3152 pts/1    Ss   13:05   0:00 bash
root        53  0.0  0.0  36632  2748 pts/1    R+   13:05   0:00 ps -aux
root@sonic:/# supervisorctl status
bgpcfgd                          RUNNING   pid 34, uptime 0:02:00
bgpd                             RUNNING   pid 39, uptime 0:01:59
fpmsyncd                         RUNNING   pid 45, uptime 0:01:59
rsyslogd                         RUNNING   pid 29, uptime 0:02:00
start.sh                         EXITED    May 17 01:03 PM
vtysh_b                          STOPPED   Not started
zebra                            RUNNING   pid 37, uptime 0:01:59

**Comment by nazariig** [comment]
#2914

**Comment by pavel-shirshov** [comment]
@nazariig So it wasn't about staticd, but about swig. I have approved your PR. I'll push it as soon as mellonox build is done.
